### ISSUE TITLE
Injecting LogPlugin twice causes a panic

### ISSUE THREAD

**Issue by lecoqjacob** [bevyengine/bevy#6426]
## Bevy version
Main branch (0.9-dev)

## \[Optional\] Relevant system information

**Specs**: 14-in M1 Macbook Pro running 12.6 Monterey, with 32 GB Ram

## What you did

Injecting `DefaultPluigins` while also modifying the `LogPlugin`, works as expected.

```rust
DefaultPlugins
            .set(WindowPlugin {
                window: WindowDescriptor {
                    width: WINDOW_WIDTH,
                    height: WINDOW_HEIGHT,
                    title: String::from(LAUNCHER_TITLE),
                    // present_mode: bevy::window::PresentMode::AutoVsync,
                    ..default()
                },
                ..default()
            })
            .set(ImagePlugin::default_nearest())
            .set(bevy::log::LogPlugin { ..default() });
```

Injecting a second `LogPlugin` causes a panic for bevy::log/src/lib.rs:118:27

```rust
app.add_plugin(bevy::log::LogPlugin { ..default() });
```

Only when I disable the log plugin from the initial `DefaultPlugins` does it actually allow me to inject a second context.

```rust
defaults.build().disable::<bevy::log::LogPlugin>()
...

// Yay this works :)
app.add_plugin(bevy::log::LogPlugin { ..default() });
```

## Additional information

Here is the full stack trace:

```bash
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: SetLoggerError(())', /Users/elkxj/.cargo/git/checkouts/bevy-f7ffde730c324c74/7540328/crates/bevy_log/src/lib.rs:118:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by mockersf** [comment]
related to #1969
panic would change with #6411

**Comment by alice-i-cecile** [comment]
Fixed by #6411.

**Comment by lecoqjacob** [comment]
You guys are the best! Thanks!
